docs/extend/creating-classic-plugins.md

```diff
@@ -41,28 +41,130 @@ These examples provide the bare bones needed to get started. For more informatio
+  - manage_threads
+  - set_https_connection_properties
+  - outbound_network
+```
+### Entitlements
+
+The entitlements currently implemented and enforced in {{es}} that are available to plugins are the following ones:
+
+#### `manage_threads`
+
+Allows code to call methods that create or modify properties on Java Threads, for example `Thread#start` or `ThreadGroup#setMaxPriority`. In general, setting the name, priority, daemon state and context class loader are things no plugins should do when executing on
+{{es}} threadpools; however, many 3rd party libraries that support async operations (e.g. Apache HTTP client) need to manage their own threads. In this case it is justifiable to request this entitlement.
 
-SecurityManager sm = System.getSecurityManager();
-if (sm != null) {
-  // unprivileged code such as scripts do not have SpecialPermission
-  sm.checkPermission(new SpecialPermission());
-}
-AccessController.doPrivileged(
-  // sensitive operation
-);
+Example:
+```yaml
+org.example.module: # or 'ALL-UNNAMED' if the plugin is non-modular
+  - manage_threads
 ```
 
-Check [Secure Coding Guidelines for Java SE](https://www.oracle.com/technetwork/java/seccodeguide-139067.md) for more information.
+#### `outbound_network`
+
+Allows code to call methods to connect to an endpoint. {{es}} does not grant any network access by default; each plugin that needs to directly connect to an external resource (e.g. to upload or download data) must request this entitlement.
+
+Example:
+```yaml
+org.example.module: # or 'ALL-UNNAMED' if the plugin is non-modular
+  - outbound_network
+```
 
+#### `set_https_connection_properties`
+Allows code to call methods to change properties on an established HTTPS connection. While this is generally innocuous (e.g. the google API client uses it to modify the HTTPS connections they just created), these methods can allow code to change arbitrary connections.
 
+Example:
+```yaml
+org.example.module: # or 'ALL-UNNAMED' if the plugin is non-modular
+  - set_https_connection_properties
+```
+
+#### `inbound_network` (deprecated)
+Allows code to call methods to listen for incoming connections, so external resources can connect directly to your plugin. This entitlement should only be used when absolutely necessary (e.g. if a library you depend on requires it for authentication). Granting it makes the {{es}} node more vulnerable to attacks. This entitlement is deprecated, and can be removed in a future version of {{es}}.
+
+Example:
+```yaml
+org.example.module: # or 'ALL-UNNAMED' if the plugin is non-modular
+  - inbound_network
+```
+
+#### `load_native_libraries`
+Allows code to load native libraries and call restricted methods. This entitlement also enables native access for modules it is granted to (see [Restricted Methods](https://docs.oracle.com/en/java/javase/24/core/restricted-methods.html) in the Java documentation). Native code may alter the JVM or circumvent access checks such as file or network restrictions.
```

Comment:
This "see" part can be simplified by just making the phrase _restricted methods_ a link the first time it occurs.

Reply:
IDK, the page contains information about both restricted methods and what "native access" means and how this is needed on a by-module base.. So I wanted the link to be in that position to "cover" both restricted methods and native access. But probably it's difficult to see anyway :)